pkg/metrics/metrics.go

```diff
@@ -67,57 +118,180 @@ func init() {
 	metricUpgradeStartTime.With(prometheus.Labels{
-		nameLabel: upgradeconfig}).Set(
-			float64(time.Unix()))
+		versionLabel: version,
+		nameLabel:    upgradeConfigName}).Set(
+		float64(time.Unix()))
 }
 
-func (c *Counter) UpdateMetricControlPlaneEndTime(time time.Time, upgradeconfig string) {
+func (c *Counter) UpdateMetricControlPlaneEndTime(time time.Time, upgradeConfigName string, version string) {
 	metricControlPlaneUpgradeEndTime.With(prometheus.Labels{
-		nameLabel: upgradeconfig}).Set(
-			float64(time.Unix()))
+		versionLabel: version,
+		nameLabel:    upgradeConfigName}).Set(
+		float64(time.Unix()))
+}
+
+func (c *Counter) IsMetricUpgradeStartTimeSet(upgradeConfigName string, version string) (bool, error) {
+	cpMetrics, err := c.Query(fmt.Sprintf("upgrade_start_timestamp{%s=\"%s\",%s=\"%s\"", nameLabel, upgradeConfigName, versionLabel, version))
+	if err != nil {
+		return false, err
+	}
+
+	if len(cpMetrics.Data.Result) > 0 {
+		return true, nil
+	}
+
+	return false, nil
+}
+
+func (c *Counter) IsMetricControlPlaneEndTimeSet(upgradeConfigName string, version string) (bool, error) {
+	cpMetrics, err := c.Query(fmt.Sprintf("controlplane_upgrade_end_timestamp{%s=\"%s\",%s=\"%s\"", nameLabel, upgradeConfigName, versionLabel, version))
+	if err != nil {
+		return false, err
+	}
+
+	if len(cpMetrics.Data.Result) > 0 {
+		return true, nil
+	}
+
+	return false, nil
+}
+
+func (c *Counter) IsMetricNodeUpgradeEndTimeSet(upgradeConfigName string, version string) (bool, error) {
+	cpMetrics, err := c.Query(fmt.Sprintf("node_upgrade_end_timestamp{%s=\"%s\",%s=\"%s\"", nameLabel, upgradeConfigName, versionLabel, version))
+	if err != nil {
+		return false, err
+	}
+
+	if len(cpMetrics.Data.Result) > 0 {
+		return true, nil
+	}
+
+	return false, nil
 }
 
-func (c *Counter) UpdateMetricNodeUpgradeEndTime(time time.Time, upgradeconfig string) {
+func (c *Counter) UpdateMetricNodeUpgradeEndTime(time time.Time, upgradeConfigName string, version string) {
 	metricNodeUpgradeEndTime.With(prometheus.Labels{
-		nameLabel: upgradeconfig}).Set(
-			float64(time.Unix()))
+		versionLabel: version,
+		nameLabel:    upgradeConfigName}).Set(
+		float64(time.Unix()))
 }
 
-func (c *Counter) UpdateMetricClusterVerificationFailed(upgradeconfig string) {
+func (c *Counter) UpdateMetricClusterVerificationFailed(upgradeConfigName string) {
 	metricClusterVerificationFailed.With(prometheus.Labels{
-		nameLabel: upgradeconfig}).Set(
-			float64(1))
+		nameLabel: upgradeConfigName}).Set(
+		float64(1))
 }
 
-func (c *Counter) UpdateMetricClusterVerificationSucceeded(upgradeconfig string) {
+func (c *Counter) UpdateMetricClusterVerificationSucceeded(upgradeConfigName string) {
 	metricClusterVerificationFailed.With(prometheus.Labels{
-		nameLabel: upgradeconfig}).Set(
-			float64(0))
+		nameLabel: upgradeConfigName}).Set(
+		float64(0))
+}
+
+func getPromUrl(c client.Client) (*string, error) {
```

Comment:
Minor nit.. when I think 'url' I think of the form with the protocol and path included, eg 'https://blah'.. maybe rename to getPromHost() to be clearer about what it's returning?

Reply:
:+1: